albumentations/augmentations/functional.py

```diff
@@ -418,6 +418,62 @@ def shift_hsv(img, hue_shift, sat_shift, val_shift):
     return img
 
 
+def _shift_color_space_uint8(img, shifts, limits, code, is_angle):
+    img = cv2.cvtColor(img, code[0])
+    img = cv2.split(img)
+
+    for channel, is_a, shift, limit in zip(img, is_angle, shifts, limits):
+        if shift == 0:
+            continue
+
+        lut = np.arange(0, 256, dtype=np.int16)
+        if is_a:
+            lut = np.mod(lut + shift, limit[1]).astype(np.uint8)
+        else:
+            lut = np.clip(lut + shift, limit[0], limit[1]).astype(np.uint8)
+
+        cv2.LUT(channel, lut, dst=channel)
+
+    img = cv2.merge(img)
```

Comment:
Can we avoid cv2.split/cv2.merge by doing cv2.LUT inplace?  `cv2.LUT(img[..., channel], lut, dst=img[..., channel])` ?

Reply:
This doesn't work, because numpy is creating a new array.

Reply:
This version is slower.
Current version: 120 FPS
`img[..., i] = cv2.LUT(img[..., i], lut)`: 110 FPS

Reply:
Version `img[..., i] = lut[img[..., i]]` gives 60 FPS